src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -86,18 +114,21 @@ private[spark] object Config {
           "Specifically, if set, all gang jobs are scheduled onto nodes for which the value of the provided label is equal."
       )
       .stringConf
-      .checkValue(K8sValidator.Label.isValidKey, "Only a valid RFC 1123 label key is allowed!")
-      .createWithDefaultString("armada-spark")
+      .checkValue(
+        v => if (v.nonEmpty) K8sValidator.Label.isValidKey(v) else true,
+        "Only a valid RFC 1123 label key is allowed!"
+      )
+      .createWithDefaultString("")
```

Comment:
```suggestion
      .stringConf
      .createOptional
```
Please review all other options whether `""` is really a meaningful default value (it is if the string is used literally).